### ISSUE TITLE
Implement feature extractor

### ISSUE THREAD

**Issue by sarambulo** [nikitachaudharicodes/ProjeXion#3]
Create a `feature_extraction.py` module inside the `models` directory that implements the class `ImageEncoder`. This class should inherit from `torch.nn.Module` and should take as input a HxWx3 image and generate an output embedding of size H/4xW/4x32 using 8 2D-CNN layers:

*An eight-layer 2D CNN is applied, where the strides of layer 3 and 6 are set to two to divide the feature towers into three scales. Within each scale, two convolutional layers are applied to extract the higher-level image representation. Each convolutional layer is followed by a batch-normalization (BN) layer and a rectified linear unit (ReLU) except for the last layer. Also, similar to common matching tasks, parameters are shared among all feature towers for eficient learning. The outputs of the 2D network are N 32-channel feature maps downsized by four in each dimension compared with input images.*

![Image](https://github.com/user-attachments/assets/4902ac84-7a30-414e-b05b-a72c738ac877)

**Comment by maitribg** [comment]
Done!
